### DISCUSSION TITLE
How can I export primevue datatable as PDF?

### DISCUSSION THREAD

**Discussion by oguzhanylcnkaya** [primefaces/community#256] [PrimeVue] [answered]
When I read the primevue datatable document, I saw that it can only be downloaded in CSV format. Does it allow downloading in PDF format?

**Comment by oguzhanylcnkaya** [comment, marked as answer]
I saw how it was used in Prime React and adapted it in Vue

https://primereact.org/datatable/#export

**Reply by anilrepala505** [reply]
Can you share the code how to export datatable as PDF in primevue?

**Reply by LeoArurault** [reply]
A bit old but the code can still be usefull to someone else.

You will have to add `jspdf` and `jspdf-autotable` as dependencies.

```typescript
import jsPDF from 'jspdf';
import autoTable from 'jspdf-autotable';

const dt = ref();

const exportPdf = () => {
    const doc = new jsPDF();
    autoTable(doc, { html: dt.value.$refs.table });
    doc.save('filename.pdf');
};
```

```html
<DataTable ref="dt">...</>
<Button icon="pi pi-file-pdf" label="Export PDF" @click="exportPdf()" />

```
